ca/ca_test.go

```diff
@@ -603,71 +630,69 @@ func countMustStaple(t *testing.T, cert *x509.Certificate) (count int) {
+	test.AssertEquals(t, countMustStaple(t, cert), 1)
+}
 
+func issueCertificateSubTestDuplicateMustStaple(t *testing.T, ca *CertificateAuthorityImpl, _ *mockSA) {
 	// Even if there are multiple TLS Feature extensions, only one extension should be included
-	duplicateMustStapleCert := sign(DuplicateMustStapleCSR)
+
+	coreCert, err := ca.IssueCertificate(ctx, &caPB.IssueCertificateRequest{Csr: DuplicateMustStapleCSR, RegistrationID: &arbitraryRegID})
+	test.AssertNotError(t, err, "Failed to issue")
+	cert, err := x509.ParseCertificate(coreCert.DER)
+	test.AssertNotError(t, err, "Error parsing certificate produced by CA")
+
 	test.AssertEquals(t, count(csrExtensionCategory, csrExtensionTLSFeature, ca.csrExtensionCount), 1)
 	test.AssertEquals(t, count(csrExtensionCategory, csrExtensionTLSFeatureInvalid, ca.csrExtensionCount), 0)
 	test.AssertEquals(t, signatureCountByPurpose("cert", ca.signatureCount), 1)
-	test.AssertEquals(t, countMustStaple(t, duplicateMustStapleCert), 1)
+	test.AssertEquals(t, countMustStaple(t, cert), 1)
+}
 
-	// ... but if it doesn't ask for stapling, there should be an error
-	ca.csrExtensionCount.Reset()
-	ca.signatureCount.Reset()
-	_, err = ca.IssueCertificate(ctx, &caPB.IssueCertificateRequest{Csr: TLSFeatureUnknownCSR, RegistrationID: &arbitraryRegID})
+func issueCertificateSubTestTLSFeatureUnknown(t *testing.T, ca *CertificateAuthorityImpl, _ *mockSA) {
 	test.AssertEquals(t, count(csrExtensionCategory, csrExtensionTLSFeature, ca.csrExtensionCount), 1)
 	test.AssertEquals(t, count(csrExtensionCategory, csrExtensionTLSFeatureInvalid, ca.csrExtensionCount), 1)
-	test.AssertEquals(t, signatureCountByPurpose("cert", ca.signatureCount), 0)
-	test.AssertError(t, err, "Allowed a CSR with an empty TLS feature extension")
-	test.Assert(t, berrors.Is(err, berrors.Malformed), "Wrong error type when rejecting a CSR with empty TLS feature extension")
+}
+
+func issueCertificateSubTestUnknownExtension(t *testing.T, ca *CertificateAuthorityImpl, _ *mockSA) {
+	// Unsupported extensions in the CSR should be silently ignored.
+
+	coreCert, err := ca.IssueCertificate(ctx, &caPB.IssueCertificateRequest{Csr: UnsupportedExtensionCSR, RegistrationID: &arbitraryRegID})
+	test.AssertNotError(t, err, "Failed to issue")
+	cert, err := x509.ParseCertificate(coreCert.DER)
+	test.AssertNotError(t, err, "Error parsing certificate produced by CA")
 
-	// Unsupported extensions should be silently ignored, having the same
-	// extensions as the TLS Feature cert above, minus the TLS Feature Extension
-	unsupportedExtensionCert := sign(UnsupportedExtensionCSR)
 	test.AssertEquals(t, count(csrExtensionCategory, csrExtensionOther, ca.csrExtensionCount), 1)
 	test.AssertEquals(t, signatureCountByPurpose("cert", ca.signatureCount), 1)
-	test.AssertEquals(t, len(unsupportedExtensionCert.Extensions), len(singleStapleCert.Extensions)-1)
+
+	// XXX: The hard-coded value here will have to change over time as
```

Comment:
In this codebase, we'd consider this a NOTE. We generally use XXX for "remind me to fix this before merging."